aces/Loris
# [New Feature] Tracking and Configuration of Candidate's Diagnosis Evolution

## Brief summary of changes
This PR adds a way for project administrators to define and configure diagnosis trajectories for their project. The aim of this new feature is multifaceted; it allows study coordinators to 
1. Track the evolution of a patient's diagnosis over time
2. Determine (& query) the candidate's latest diagnosis is without having to manually compare the longitudinal data

**_Interest for this feature was expressed by multiple projects during the DCC meeting on July 7th._**
 ______

### What is included in this PR?
**1. A configuration page for diagnosis evolution** 

<img width="1423" alt="Screen Shot 2021-08-31 at 2 50 15 PM" src="https://user-images.githubusercontent.com/34260251/131567053-84acf7a8-892e-4e30-bec5-99ddaf61078f.png">


**2. A tab in candidate parameters to view details on the evolution of the diagnosis.**

<img width="1265" alt="Screen Shot 2021-08-31 at 2 51 17 PM" src="https://user-images.githubusercontent.com/34260251/131567148-115d05dd-2e99-4ae7-affa-eb386f8ea74e.png">

**3. Logic to propagate the latest diagnosis data as instruments are updated.
4. A script that all projects should run when configuring Dx Trajectories and updating their instances.
5. Updates the demographic import to couchDB so the latest diagnosis can be queried.**


_____

### Info
A `Diagnosis Trajectory` is defined by a source field, or group of source fields, that are found within the test battery.
A patient's `Diagnosis Evolution` is defined by the progression of changing diagnoses over time. The evolution is determined by the order of configured `Diagnosis Trajectories`. 

There are two ways that the `Latest Diagnosis` of the candidate can be updated. First, the script in this PR can be run. Secondly, when ANY instrument part of a configured diagnosis track is saved, the latest diagnosis is determined and updated in the `candidate` table.  

EX:

<img width="1172" alt="Screen Shot 2021-08-31 at 3 29 58 PM" src="https://user-images.githubusercontent.com/34260251/131568429-eddcd52f-9dad-4523-891b-cc898392ecba.png">

<img width="1434" alt="Screen Shot 2021-08-31 at 3 16 59 PM" src="https://user-images.githubusercontent.com/34260251/131568452-90fe1b67-85d5-4867-9941-9646c74f01e7.png">







#### Testing instructions (if applicable)

1. `fetch jesscall && git checkout jesscall/2021_07_27_diagnosis_evolution`
2. `cat SQL/New_patches/2021-07-28_diagnosis_evolution.sql | mysql`
3. `make dev`
4. Follow instructions in configuration & caniddate parameters TestPlan.md files included in this PR
5. **After setting up the configuration for a diagnosis evolution please do the following:**
6. `php tools/data_integrity/update_candidate_latest_diagnosis.php`
7. Ensure that updating data in a configured instrument updates the latest diagnosis in the `candidate` table (if it's the latest) and the candidate parameters tab.

#### Link(s) to related PR(s)
* Blocked by #7514
